nifi-nar-bundles/nifi-extension-utils/nifi-hadoop-utils/src/main/java/org/apache/nifi/processors/hadoop/AbstractHadoopProcessor.java

```diff
@@ -216,21 +230,22 @@ protected KerberosProperties getKerberosProperties(File kerberosConfigFile) {
                     .build());
         }
 
-        if (credentialsService != null && (explicitPrincipal != null || explicitKeytab != null)) {
+        if (credentialsService != null && (explicitPrincipal != null || explicitKeytab != null || explicitPassword != null)) {
             results.add(new ValidationResult.Builder()
                 .subject("Kerberos Credentials")
                 .valid(false)
-                .explanation("Cannot specify both a Kerberos Credentials Service and a principal/keytab")
+                .explanation("Cannot specify a Kerberos Credentials Service while also specifying a Kerberos Principal, Kerberos Keytab, or Kerberos Password")
                 .build());
         }
 
         final String allowExplicitKeytabVariable = System.getenv(ALLOW_EXPLICIT_KEYTAB);
-        if ("false".equalsIgnoreCase(allowExplicitKeytabVariable) && (explicitPrincipal != null || explicitKeytab != null)) {
+        if ("false".equalsIgnoreCase(allowExplicitKeytabVariable) && (explicitPrincipal != null || explicitKeytab != null || explicitPassword != null)) {
```

Comment:
I'm not sure if we want to include password into this logic. If ALLOW_EXPLICIT_KEYTAB=false, it seems valid to still use principal+password.

Reply:
Updated the code to allow that scenario, and added a unit test to verify the validation logic in AbstractHadoopProcessor.  This works in conjunction with the added test cases in TestKerberosProperties  to validate principal/keytab/password combinations.